hash/hash.go

```diff
@@ -0,0 +1,101 @@
+// Package hash provides utility functions to process complex data types, like
+// data streams, files, or sequences of structs of different types.
+package hash
+
+import (
+	"bytes"
+	"errors"
+	h "hash"
```

Comment:
style: I'd prefer avoiding aliasing packages, it makes it more difficult to understand code and compare it to other places where the package keeps the `hash` name.